src/Amalgam/interpreter/Interpreter.h

```diff
@@ -786,44 +866,39 @@ class Interpreter
 		return true;
 	}
 
-	//recalculates curNumExecutionNodes
-	__forceinline void UpdateCurNumExecutionNodes()
+	//if true, no limit on how much memory can utilize
+	constexpr bool ConstrainedAllocatedNodes()
 	{
-		curNumExecutionNodes = curNumExecutionNodesAllocatedToEntities + evaluableNodeManager->GetNumberOfUsedNodes();
+		return (performanceConstraints != nullptr && performanceConstraints->ConstrainedAllocatedNodes());
```

Comment:
shouldn't this also return true if performanceConsntrains are undefined (nullptr) as well?